homeassistant/components/permobil/coordinator.py

```diff
@@ -0,0 +1,57 @@
+"""DataUpdateCoordinator for permobil integration."""
+
+from datetime import timedelta
+import logging
+
+import async_timeout
+from mypermobil import (
+    ENDPOINT_BATTERY_INFO,
+    ENDPOINT_DAILY_USAGE,
+    ENDPOINT_VA_USAGE_RECORDS,
+    MyPermobil,
+    MyPermobilAPIException,
+    MyPermobilClientException,
+)
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import (
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MyPermobilCoordinator(DataUpdateCoordinator):
+    """MyPermobil coordinator."""
+
+    def __init__(self, hass: HomeAssistant, p_api: MyPermobil) -> None:
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="permobil",
+            update_interval=timedelta(seconds=50),
+        )
+        self.p_api = p_api
+
+    async def _async_update_data(self):
```

Comment:
We be nice is we could add a return type here, e.g., you could define a TypedDict (or even better a dataclass and use that).